Dockerfile

```diff
@@ -0,0 +1,37 @@
+FROM rust:latest as build-stage
+
+COPY Cargo.lock .
+COPY Cargo.toml .
+
+RUN apt-get -y update && apt-get -y install clang llvm pkg-config nettle-dev
+
+RUN mkdir src && \
+    echo "fn main() {print!(\"Dummy main\");} // dummy file" > src/main.rs && \
+    set -x && cargo build --release && \
+    set -x && rm target/release/deps/wkd_server*
+
+# Now add the rest of the project and build the real main
+COPY src ./src
+RUN set -x && cargo build --release
+
+# Create a minimal docker image 
+FROM debian:bullseye-slim
+
+RUN apt-get -y update && \
+    apt-get install -y \
+    libpq-dev \
+    openssl \
+    libssl-dev \
+    ca-certificates && \
+    apt-get autoremove && \
+    echo -n 'Yes, do as I say!' | apt-get remove apt && \
+    rm -rf /var
+
+ENV RUST_LOG="error,rust-wkd-server=info"
+COPY --from=build-stage /target/release/wkd-server /
+
+EXPOSE 8080
+
+VOLUME /openpgp-keys
```

Comment:
Isn't this declared in `compose.yaml`?

Reply:
depends on the compose file, afaik `VOLUME` and `EXPOSE` are merely best practices and do not really need to be there, but I would keep them.